### PR TITLE
Update JITServerRelocationRuntime to be merged into master

### PR DESCRIPTION
**Pull request by a7ehuo**
Update `JITServerRelocationRuntime` to be merged into the `master` branch.

The mirrored master change is in #7236.

Signed-off-by: Annabelle Huo <Annabelle.Huo@ibm.com>

### CODE REVIEW

**Comment by a7ehuo** [comment]
@mpirvu Ready for review. Thanks! 
This PR might have a bit overlap with #7209 on porting the removal of the unused `J9JavaVM` parameter (503e61f62) from the master branch.

File: runtime/compiler/runtime/RelocationRuntime.cpp [hunk 1]
```diff
@@ -169,7 +171,11 @@ TR_RelocationRuntime::prepareRelocateAOTCodeAndData(J9VMThread* vmThread,
    _fe = theFE;
    _codeCache = aotMCCRuntimeCodeCache;
    _method = theMethod;
+#if defined(JITSERVER_SUPPORT)
```
Comment (mpirvu) [inline runtime/compiler/runtime/RelocationRuntime.cpp]: Similar to the change to master, I don't think the ifdef is needed

**Comment by a7ehuo** [comment]
Will make the update on `jitaas` branch #7236 first.

**Comment by a7ehuo** [comment]
@mpirvu All comments from #7236 are addressed for this mirrored `jitaas`  branch change as well. Ready fo review. Thanks.

**Comment by mpirvu** [comment]
Jenkins test sanity.functional+jitaas xlinux jdk8,jdk11 depends eclipse/openj9-omr#jitaas

**Comment by mpirvu** [comment]
Jenkins test sanity.functional+jitaas xlinux jdk11 depends eclipse/openj9-omr#jitaas

**Comment by mpirvu** [comment]
Jenkins test sanity.functional+jitaas xlinux jdk8 depends eclipse/openj9-omr#jitaas

**Comment by a7ehuo** [comment]
For [Java 8](https://ci.eclipse.org/openj9/job/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal/313/console) Jenkins test, `TestConfig` test build failed. Looks like the Jenkins test issue is being addressed in https://github.com/eclipse/openj9/pull/7292#issuecomment-538012699. 
```
11:00:05      [javac] /home/jenkins/workspace/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal/openjdk-tests/functional/Java8andUp/src/org/openj9/test/attachAPI/TestJmap.java:93: error: cannot access Response
11:00:05      [javac] 		InputStream result = myVm.heapHisto("-all"); //$NON-NLS-1$
...
BUILD FAILED
11:00:09  /home/jenkins/workspace/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal/openjdk-tests/TestConfig/scripts/build_test.xml:73: The following error occurred while executing this line:
11:00:09  /home/jenkins/workspace/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal/openjdk-tests/functional/build.xml:52: The following error occurred while executing this line:
...
```

For [Java 11](https://ci.eclipse.org/openj9/job/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/275/console), looks like a crash in `libj9jit29.so`. Tried to repro it locally with sanity functional test. So far haven't observed the crash yet. 

```
04:44:19  #0: /home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdkbinary/j2sdk-image/lib/compressedrefs/libj9jit29.so(+0x896835) [0x7fb66da4a835]
04:44:19  #1: /home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdkbinary/j2sdk-image/lib/compressedrefs/libj9jit29.so(+0x8a2a90) [0x7fb66da56a90]
04:44:19  #2: /home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdkbinary/j2sdk-image/lib/compressedrefs/libj9jit29.so(+0x1722ae) [0x7fb66d3262ae]
04:44:19  #3: /home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdkbinary/j2sdk-image/lib/compressedrefs/libj9prt29.so(+0x1ed7a) [0x7fb679947d7a]
04:44:19  #4: /lib64/libpthread.so.0() [0x3545a0f7e0]
04:44:19  #5: /home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdkbinary/j2sdk-image/lib/compressedrefs/libj9jit29.so(+0x21e24c) [0x7fb66d3d224c]
04:44:19  #6: /home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdkbinary/j2sdk-image/lib/compressedrefs/libj9jit29.so(+0x22f8a1) [0x7fb66d3e38a1]
04:44:19  #7: /home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdkbinary/j2sdk-image/lib/compressedrefs/libj9jit29.so(+0x210282) [0x7fb66d3c4282]
04:44:19  #8: /home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdkbinary/j2sdk-image/lib/compressedrefs/libj9jit29.so(+0x212006) [0x7fb66d3c6006]
```

**Comment by mpirvu** [comment]
`183 - - AttachAPISanity_0` seen again. Possibly changes in testing framework causes this.

**Comment by JasonFengJ9** [comment]
> 11:00:05      [javac] /home/jenkins/workspace/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal/openjdk-tests/functional/Java8andUp/src/org/openj9/test/attachAPI/TestJmap.java:93: error: cannot access Response
11:00:05      [javac] 		InputStream result = myVm.heapHisto("-all"); //$NON-NLS-1$

#7279 and #7313 need to be cherry-picked, and they depends on ibmruntimes/openj9-openjdk-jdk8#327, ibmruntimes/openj9-openjdk-jdk11#210, ibmruntimes/openj9-openjdk-jdk12#71, ibmruntimes/openj9-openjdk-jdk13#13, ibmruntimes/openj9-openjdk-jdk#138.

**Comment by mpirvu** [comment]
Since the required PRs for testing were merged into the jitaas branch, a rebase may be required to pick them up

**Comment by a7ehuo** [comment]
@mpirvu Rebased to `jitaas` branch 0dac0fa4. Ready for PR Jenkins build. Thanks!

**Comment by mpirvu** [comment]
Jenkins test sanity.functional+jitaas xlinux jdk8,jdk11 depends eclipse/openj9-omr#jitaas

**Comment by a7ehuo** [comment]
Java 8 failed with the following one test: 
```
11:02:25  FAILED test targets:
11:02:25  	cmdLineTester_gcsuballoctests_0
11:02:25  
11:02:25  TOTAL: 204   EXECUTED: 117   PASSED: 116   FAILED: 1   DISABLED: 6   SKIPPED: 81
```

Java 11 failed with the following two tests: `J9vmTest_2` could be intermittent.
```
12:22:11  FAILED test targets:
12:22:11  	cmdLineTester_gcsuballoctests_0
12:22:11  	J9vmTest_2
12:22:11  
12:22:11  TOTAL: 226   EXECUTED: 138   PASSED: 136   FAILED: 2   DISABLED: 4   SKIPPED: 84
```

**Comment by mpirvu** [comment]
I am going to issue Java11 tests again to see whether `J9vmTest_2` is solid or not.

**Comment by mpirvu** [comment]
Jenkins test sanity.functional+jitaas xlinux jdk11 depends eclipse/openj9-omr#jitaas

**Comment by mpirvu** [comment]
This time Java 11 failed
```
cmdLineTester_gcsuballoctests_0
cmdLineTester_jvmtitests_5
J9vmTest_2
```
J9vmTest_2 appears again.
```
+++ j9vm.test.jni.CriticalRegionTest: +++
command: /home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdkbinary/j2sdk-image/bin/java -XX:+UseJITServer -Xcompressedrefs -Xgcpolicy:metronome -Xcompressedrefs  -Xdump -Xms64m -Xmx64m    -classpath /home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdk-tests/TestConfig/scripts/testKitGen/../../../../jvmtest/functional/VM_Test/VM_Test.jar:/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Personal/openjdk-tests/TestConfig/scripts/testKitGen/../../../../jvmtest/TestConfig/lib/asm-all.jar  j9vm.test.jni.CriticalRegionTest 

Exception in thread "main" java.lang.BootstrapMethodError: java.lang.invoke.StringConcatException: Generator failed
	at j9vm.test.jni.CriticalRegionTest.testAcquireAndSleep(CriticalRegionTest.java:756)
	at j9vm.test.jni.CriticalRegionTest.runBlockingTests(CriticalRegionTest.java:733)
	at j9vm.test.jni.CriticalRegionTest.main(CriticalRegionTest.java:191)
Caused by: java.lang.invoke.StringConcatException: Generator failed
	at java.base/java.lang.invoke.StringConcatFactory.generate(StringConcatFactory.java:764)
	at java.base/java.lang.invoke.StringConcatFactory.doStringConcat(StringConcatFactory.java:665)
	at java.base/java.lang.invoke.StringConcatFactory.makeConcatWithConstants(StringConcatFactory.java:581)
	at java.base/java.lang.invoke.MethodHandle.invokeBsm(MethodHandle.java:963)
	at java.base/java.lang.invoke.MethodHandle.resolveInvokeDynamic(MethodHandle.java:1093)
	... 3 more
Caused by: java.lang.ClassCastException: [Ljava.lang.Class; incompatible with java.util.concurrent.ConcurrentHashMap$Node
	at java.base/java.util.concurrent.ConcurrentHashMap.tabAt(ConcurrentHashMap.java:760)
	at java.base/java.util.concurrent.ConcurrentHashMap.get(ConcurrentHashMap.java:938)
	at java.base/java.lang.invoke.ThunkTable.get(MethodHandle.java:1704)
	at java.base/java.lang.invoke.FoldHandle.computeThunks(FoldHandle.java:69)
	at java.base/java.lang.invoke.MethodHandle.<init>(MethodHandle.java:235)
	at java.base/java.lang.invoke.FoldHandle.<init>(FoldHandle.java:33)
	at java.base/java.lang.invoke.FoldNonvoidHandle.<init>(FoldHandle.java:104)
	at java.base/java.lang.invoke.FoldHandle.get(FoldHandle.java:55)
	at java.base/java.lang.invoke.MethodHandles.foldArgumentsCommon(MethodHandles.java:2721)
	at java.base/java.lang.invoke.MethodHandles.foldArguments(MethodHandles.java:2637)
	at java.base/java.lang.invoke.StringConcatFactory$MethodHandleInlineCopyStrategy.generate(StringConcatFactory.java:1560)
	at java.base/java.lang.invoke.StringConcatFactory.generate(StringConcatFactory.java:756)
	... 7 more
no-zero exit value: 1
*** Test FAILED *** (j9vm.test.jni.CriticalRegionTest)
```

**Comment by mpirvu** [comment]
Jenkins test sanity.functional+jitaas xlinux jdk11 depends eclipse/openj9-omr#jitaas

**Comment by mpirvu** [comment]
This time only the known `cmdLineTester_gcsuballoctests_0` test failed.

**Pull request merged**